test/functional/e2e_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"flag"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	resmgr "github.com/intel/cri-resource-manager/pkg/cri/resource-manager"
+	"google.golang.org/grpc"
+	api "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+const (
+	testDir = "/tmp/cri-rm-test"
+)
+
+func setUp(t *testing.T) {
+	t.Helper()
+
+	if err := os.MkdirAll(testDir, 0700); err != nil {
```

Comment:
Should we use `ioutil.TempDir` or smth to avoid possible races?

Reply:
Agree. Even though mkdir is an atomic syscall there's no need to call it for every test function. I put it to `init()`.